src/cmd/tools/read_data_files/main/main.go

```diff
@@ -248,20 +254,63 @@ func main() {
 	}
 }
 
+func getVolumeNumber(pathPrefix string, namespace string, shard int, blockStart string) (int, error) {
+	nsID := ident.StringID(namespace)
+	path := fs.NamespaceDataDirPath(pathPrefix, nsID)
+	files, err := ioutil.ReadDir(path + "/" + strconv.Itoa(shard))
+	if err != nil {
+		return 0, fmt.Errorf("failed reading data directory: %w", err)
+	}
+
+	volumeSet := make(map[int]struct{})
+	maxVolumeNum := 0
+	for _, f := range files {
+		if f.IsDir() {
+			continue
+		}
+		splittedFileName := strings.Split(f.Name(), "-")
+		if len(splittedFileName) != fs.MaxDelimNum {
+			fmt.Errorf("failed parsing volume number from file name: %s, not enough segments", f.Name())
+		} else {
+			if splittedFileName[1] == blockStart {
+				volumeNum, err := strconv.Atoi(splittedFileName[2])
```

Comment:
There is a potential data race. M3DB may be merging data file with volume index `-0-` with some writes at this moment. read_data_file sees `0`, but when it actually reads the data file, it's already a new file with volume index `-1-`.
Not a blocker, but it's worth keeping in mind.

Reply:
This is a good point, I think this data race is hard to foreseen when running the job unfortunately by this design.